General/CodeSmells/Comments/Src/Comments.Problem/Good/NotEnoughComments.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Comments.Problem.Good
+{
+    /// <summary>
+    /// Solution to <see cref="Bad.Sorting"/>
+    /// In algorithms or other complex functions, splitting code made not make as much sense.
+    /// It's really hard to express yourself with self documenting code when implementing algorithms.
+    /// In those cases code is not enough and a comment is needed to explain the logic better.
+    /// </summary>
+    public class Sorting
+    {
+        public static List<int> MergeSort(List<int> unsorted)
+        {
+            if (unsorted.Count <= 1)
+                return unsorted;
+
+            List<int> left = new List<int>();
+            List<int> right = new List<int>();
+
+            int middle = unsorted.Count / 2;
+            for (int i = 0; i < middle; i++)
+            {
+                left.Add(unsorted[i]);
+            }
+            for (int i = middle; i < unsorted.Count; i++)
+            {
+                right.Add(unsorted[i]);
+            }
+
```

Comment:
Wouldn't this be better in a single while loop? 
int ctr = 0
while { left.Count + right.Count <= unsorted.Count }
{
    left.Add(i);
    if (unsorted.Count % 2 == 0 && i <= unsorted.Count / 2)
    right.Add(unsorted.Count(i)
}